src/components/combobox/combobox.tsx

```diff
@@ -506,6 +512,19 @@ export class Combobox
   //
   // --------------------------------------------------------------------------
 
+  private clearValue(): void {
+    this.ignoreSelectedEventsFlag = true;
```

Comment:
@driskull @macandcheese Do you have a suggestions on how we could generalize this? There's a few places where bits and pieces are reused, but I haven't found a pattern. 😵

Reply:
Maybe `combobox-item` needs an (undocumented?) method that could be called where `calciteComboboxItemChange` isn't fired.

I agree that these flags are super hacky looking and prone to errors. We should try to come up with a pattern where we're more deliberate with what we're trying to accomplish.

Reply:
I don't think they're hacky, tbh. Sometimes we need to update props internally. We should definitely spend some time to develop a pattern for components to adopt. It's a bit sporadic at the moment.